src/client/pfs.go

```diff
@@ -1062,6 +1069,25 @@ func (c *putFileClient) PutFileURL(repoName string, commitID string, path string
 	return nil
 }
 
+func (c *putFileClient) DeleteFile(repoName string, commitID string, path string) (retErr error) {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.oneoff {
```

Comment:
That's a good question, I'll investigate and see if that's a bug.

Reply:
Both of those use `putFileWriteCloser`s which handles the one-off semantics on `Close` so they don't need to, this call does because it's not wrapping anything.